internal/controller/modelregistry_controller_status.go

```diff
@@ -0,0 +1,393 @@
+	if modelRegistry.Spec.Istio != nil {
+		// set Istio available condition
+		if !r.SetIstioCondition(ctx, req, modelRegistry) {
+			status = metav1.ConditionFalse
+			reason = ReasonResourcesUnavailable
+			message = "Istio resources for model registry %s are not available"
+		}
+
+		// set Gateway available condition
+		if modelRegistry.Spec.Istio.Gateway != nil {
+			if !r.SetGatewayCondition(ctx, req, modelRegistry) {
+				status = metav1.ConditionFalse
+				reason = ReasonResourcesUnavailable
+				message = "Istio Gateway resources for model registry %s are not available"
+			}
+		} else {
+			meta.RemoveStatusCondition(&modelRegistry.Status.Conditions, ConditionTypeGateway)
+		}
+	} else {
+		meta.RemoveStatusCondition(&modelRegistry.Status.Conditions, ConditionTypeIstio)
+		meta.RemoveStatusCondition(&modelRegistry.Status.Conditions, ConditionTypeGateway)
+	}
+
+	return status, reason, message
+}
+
+func (r *ModelRegistryReconciler) SetIstioCondition(ctx context.Context, req ctrl.Request,
+	modelRegistry *modelregistryv1alpha1.ModelRegistry) bool {
+
+	log := klog.FromContext(ctx)
+
+	reason := ReasonResourcesCreated
+	message := "Istio resources for model registry %s were successfully created"
+
+	available := true
+	// verify that virtualservice, destinationrule, authorizationpolicy are available
+	name := req.NamespacedName
+	message, available = r.CheckIstioResourcesAvailable(ctx, name, log, message, available)
+
+	message, available, reason = r.CheckAuthConfigCondition(ctx, name, log, message, available, reason)
+
+	status := metav1.ConditionFalse
+	if available {
+		if reason == ReasonResourcesAvailable {
+			status = metav1.ConditionTrue
+		}
+		// additionally verify that Deployment pod has 3 containers including the istio-envoy proxy
+		message, reason, status = r.CheckDeploymentPods(ctx, name, log, message, reason, status)
+	} else {
+		status = metav1.ConditionFalse
+		reason = ReasonResourcesUnavailable
+	}
+	meta.SetStatusCondition(&modelRegistry.Status.Conditions, metav1.Condition{Type: ConditionTypeIstio,
+		Status: status, Reason: reason,
+		Message: fmt.Sprintf(message, modelRegistry.Name)})
+
+	return status == metav1.ConditionTrue
+}
+
+func (r *ModelRegistryReconciler) CheckDeploymentPods(ctx context.Context, name types.NamespacedName,
+	log logr.Logger, message string, reason string, status metav1.ConditionStatus) (string, string, metav1.ConditionStatus) {
+
+	pods := corev1.PodList{}
+	if err := r.Client.List(ctx, &pods,
+		client.MatchingLabels{"app": name.Name, "component": "model-registry"},
+		client.InNamespace(name.Namespace)); err != nil {
+
+		log.Error(err, "Failed to get model registry pods", "name", name)
+		message = fmt.Sprintf("Failed to find Pods for model registry %%s: %s", err.Error())
+		reason = ReasonResourcesUnavailable
+		status = metav1.ConditionFalse
+
+	} else {
+		// check that pods have 3 containers
+		for _, pod := range pods.Items {
+			if len(pod.Spec.Containers) != 3 {
```

Comment:
Instead of checking for 3 pods, wouldn't be more robust to check for a pod named envoy or with a istio-specific label?

Reply:
The pod name is kinda an istio implementation detail. And with the tight control of the deployment by the model registry, this is the easiest check to look for the envoy proxy injection.